### PR TITLE
extended logging of aws authorization error

### LINKED ISSUE #245: Add hints if authentication fails
### Use Case
If authentication fails apart from the aws built-in error it will be nice to have some hints, for example print if cloudquery detected `AWS_ACCESS_KEY_ID` or `AWS_SECRET_ACCESS_KEY` or `AWS_PROFILE`.

### Additional context
This is especially useful for debugging when you set-up cloudquery in GitHub action and you are trying to understand if it's an issue with the action configuration or with cloudquery configuration.

### PR DESCRIPTION
**Pull request by amanenk**
fixes #245

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -481,3 +487,24 @@ func obfuscateAccountId(accountId string) string {
...
+		"AWS_ROLE_ARN",
+		"AWS_SESSION_TOKEN",
+		"AWS_SHARED_CREDENTIALS_FILE",
+	}
+	var result []string
+	for _, v := range vars {
+		_, present := os.LookupEnv(v)
+		if present {
+			result = append(result, v)
+		}
```
Comment (roneli) [inline client/client.go:507]:
```suggestion
		if _, present := os.LookupEnv(v); present {
			result = append(result, v)
		}
```

File: client/client.go [hunk 2]
```diff
@@ -481,3 +487,24 @@ func obfuscateAccountId(accountId string) string {
...
+func checkEnvVariables() string {
+	vars := []string{
+		"AWS_PROFILE",
+		"AWS_ACCESS_KEY_ID",
+		"AWS_SECRET_ACCESS_KEY",
+		"AWS_CONFIG_FILE",
+		"AWS_ROLE_ARN",
+		"AWS_SESSION_TOKEN",
+		"AWS_SHARED_CREDENTIALS_FILE",
+	}
```
Comment (roneli) [inline client/client.go:501]: maybe move this up top of file ?

**Review by roneli — CHANGES_REQUESTED** [review]
LGTM, minor changes, would also ask @yevgenypats to review the message if its clearer now

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Looks good. One question :

What about if it’s not in an environment variable but just in `~/.aws/credentials`  ?

**Comment by amanenk** [comment]
> Looks good. One question :
> 
> What about if it’s not in an environment variable but just in `~/.aws/credentials` ?

there is an `creds.Source` that tell which authorization way is used.
here is an example of message below with `authorization soruce: SharedConfigCredentials: C:\\Users\\user\\.aws\\credentials`
```
{"level":"error","alias":"aws","provider":"aws","version":"unmanaged","error":"rpc error: code = Unknown desc = failed to create aws client for account default. AWS Error: operation error STS: GetCallerIdentity, https response error StatusCode: 403, RequestID: 388b55a6-781e-41fe-bef5-15b0e1121cac, api error SignatureDoesNotMatch: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details., authorization soruce: SharedConfigCredentials: C:\\Users\\user\\.aws\\credentials, detected aws env variables: AWS_PROFILE","time":"2021-12-13T16:23:34+02:00","message":"failed to configure provider"}
```

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**